AsyncMux/Sources/MuxDB.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  MuxDB.swift
+//  AsyncMux
+//
+//  Created by Lissy Harrison on 6/5/23.
+//
+
+import Foundation
+import SQLite3
+
+class MuxDB {
+    static let shared = MuxDB()
+    
+    let SQLITE_TRANSIENT = unsafeBitCast(-1, to: sqlite3_destructor_type.self)
+    
+    // specifies to store database in Files app on user's device
+    private var dbPath = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0] as String
+    
+    private var db: OpaquePointer?
+    
+    private init() {
+        db = openDatabase()
+        createTable()
+    }
+    // create database
+    func openDatabase() -> OpaquePointer? {
+        
+        let url = NSURL(fileURLWithPath: dbPath)
+        
+        // names db
+        if let pathComponent = url.appendingPathComponent("MuxDB") {
+            let filePath = pathComponent.path
+            if sqlite3_open(filePath, &db) == SQLITE_OK {
+                return db
```

Comment:
No need to return `db` here as you are modifying a class property. I.e. `openDatabase()` should not return anything.

Reply:
I don't understand this one, can we discuss further?

Reply:
`db` is a class property meaning that all class methods have access to. So no need to return it here. In other words this function could just as well be:

```
func openDatabase() {
    sqlite3_open(filePath, &db)
    // handle errors
}
```

Reply:
But in the init, 
`private var db: OpaquePointer?
    
    private init() {
        db = openDatabase()
        createTable()
    }`

Are you saying that db never needs to be initialized?  Or that one of these functions does so implicitly?